packages/ui-causal-graph-editor/src/shared/editor-overlay/legend/legend-data.tsx

```diff
@@ -14,58 +14,98 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { DefaultTheme } from '@darajs/styled-components';
-
 import { EditorMode } from '@types';
 
-export interface LegendLineDefinition {
+export interface LegendNodeDefinition {
+    /** defines the filled color of the node symbol */
+    color?: string;
+    /** defines the border color of the node symbol */
+    highlight_color?: string;
+}
+
+export interface LegendArrowDefinition {
     /** Arrow to show at end of line */
     arrowType?: 'none' | 'normal' | 'filled' | 'empty' | 'soft';
     /** Symbol to show in the center of the arrow */
     centerSymbol?: 'none' | 'cross' | 'question' | 'bidirected';
-    /** color to represent */
-    color?: keyof DefaultTheme['colors'];
+    /** defines the filled color of the edge/arrow symbol */
+    color?: string;
     /** dashArray SVG path property - line will be dashed if specified */
     dashArray?: string;
+}
+
+/**  Union type for the 'type' property */
+type GraphSymbolType = 'none' | 'edge' | 'node';
+
+interface BaseGraphLegendDefinition {
     /** legend label */
     label?: string;
-    /** Show empty spot instead of a legend line */
-    spacer?: boolean;
+    /** legend symbol type */
+    type: GraphSymbolType;
 }
 
-const RESOLVER_LEGEND: LegendLineDefinition[] = [
+// Extend the base interface for each specific 'type' value
+interface NoneGraphLegendDefinition extends BaseGraphLegendDefinition {
+    symbol?: never;
+    type: 'none'; // No symbol for 'none' type
+}
+
+export interface EdgeGraphLegendDefinition extends BaseGraphLegendDefinition {
+    symbol?: LegendArrowDefinition;
+    type: 'edge';
+}
+
+export interface NodeGraphLegendDefinition extends BaseGraphLegendDefinition {
+    symbol?: LegendNodeDefinition;
+    type: 'node';
+}
+
+export type GraphLegendDefinition = NoneGraphLegendDefinition | EdgeGraphLegendDefinition | NodeGraphLegendDefinition;
```

Comment:
This can be simplified to just a discriminated union

```suggestion
export type GraphLegendDefinition = 
    | {
        type: 'none';
    }
    | {
        type: 'edge';
        label: string;
        symbol?: LegendArrowDefinition;
    }
    | {
        type: 'node';
        label: string;
        symbol: LegendNodeDefinition;
    };
```

Then the legend type just becomes:

```tsx
type GraphLegendType = GraphLegendDefinition['type'];
```